src/coreclr/jit/gentree.cpp

```diff
@@ -6911,9 +6911,9 @@ GenTree* Compiler::gtNewIndOfIconHandleNode(var_types indType, size_t addr, GenT
         // This indirection also is invariant.
         indNode->gtFlags |= GTF_IND_INVARIANT;
 
-        if (iconFlags == GTF_ICON_STR_HDL)
+        if ((iconFlags == GTF_ICON_STR_HDL) || (iconFlags == GTF_ICON_OBJ_HDL))
         {
-            // String literals are never null
+            // String literals and frozen type objects are never null
```

Comment:
Seems like this is dead code? This would give method table. I don't think we would go through this path for frozen objects even when getting method table.

Reply:
Good point, will remove